### ISSUE TITLE
add end-2-end tests that test core api server and plug-ins features

### ISSUE THREAD

**Issue by gfichtenholt** [vmware-tanzu/kubeapps#2881]
In addition to unit-tests we should also have end-2-end tests, where we stand up a cluster deploy carvel/flux on it test out our core and plug-in apis against that.

**Comment by gfichtenholt** [comment]
I think ideally, everybody contributing to core api server and/or plug-ins, including me, should part-own it, but I can take it if no-one else will

**Comment by absoludity** [comment]
> I think ideally, everybody contributing to core api server and/or plug-ins, including me, should part-own it, but I can take it if no-one else will

+1, with the caveat that I'd be keen to focus this iteration on getting the unit tests there (and actually running those in CI - we're currently not running them in CI nor pushing an image to dockerhub), since we've committed to that. I'll move this to next iteration discussion, but no problem if we start it early.

**Comment by gfichtenholt** [comment]
Sounds good, will work on this. I have never run end-2-end tests locally. So, as a first step, I wanted to run the end-2-end tests on my laptop. So I tried to follow end-to-end-tests.md to see what all I need to do. I found out that this document is out-of-date. There are references to puppeteer, e2e-runner.yaml does not exist, etc. I am just trying to find the easiest way to run this stuff locally so I can add a test of my own. I would like to propose that the prerequisite  to my doing that is the up-to-date and easy to follow doc. Otherwise I am looking at spending a while just to set up the environment by reading scripts, etc.
I am volunteering to verify the contents of the doc, as I go along. Thanks
